File: backend/app/schemas/analytics.py
from pydantic import BaseModel
from typing import List, Dict, Any, Optional

class DashboardSummary(BaseModel):
    total_screened: int
    high_priority: int
    medium_priority: int
    low_priority: int
    mri_candidates: int
    pet_candidates: int
    biomarker_candidates: int
    pending_review: int
    completed_review: int

class StageCount(BaseModel):
    stage: str
    count: int
    percentage: float
    candidates_awaiting_review: int

class PriorityDistribution(BaseModel):
    level: str
    count: int
    percentage: float

class FunnelStep(BaseModel):
    stage: str
    count: int
    conversion_rate: float # Percentage of total cohort reaching or recommended for this stage

class AnalyticsResponse(BaseModel):
    summary: DashboardSummary
    priority_distribution: List[PriorityDistribution]
    stage_breakdown: List[StageCount]
    funnel: List[FunnelStep]
    recommended_next_stages: List[Dict[str, Any]]
    score_histogram: List[Dict[str, Any]]
    age_distribution: List[Dict[str, Any]]
    missing_data_summary: List[Dict[str, Any]]

class ResourceCapacityConfig(BaseModel):
    mri_daily_capacity: int = 15
    pet_daily_capacity: int = 5
    biomarker_daily_capacity: int = 40

class ResourceCapacityResponse(BaseModel):
    config: ResourceCapacityConfig
    mri_demand: int
    pet_demand: int
    biomarker_demand: int
    mri_utilization_pct: float
    pet_utilization_pct: float
    biomarker_utilization_pct: float
    mri_wait_days: float
    pet_wait_days: float
    disclaimer: str = "Prototype planning capacity — does not represent real hospital operational data."

File: frontend/src/types/analytics.ts
export interface DashboardSummary {
  total_screened: number;
  high_priority: number;
  medium_priority: number;
  low_priority: number;
  mri_candidates: number;
  pet_candidates: number;
  biomarker_candidates: number;
  pending_review: number;
  completed_review: number;
}

export interface PriorityDistribution {
  level: string;
  count: number;
  percentage: number;
}

export interface StageCount {
  stage: string;
  count: number;
  percentage: number;
  candidates_awaiting_review: number;
}

export interface FunnelStep {
  stage: string;
  count: number;
  conversion_rate: number;
}

export interface AnalyticsData {
  summary: DashboardSummary;
  priority_distribution: PriorityDistribution[];
  stage_breakdown: StageCount[];
  funnel: FunnelStep[];
  recommended_next_stages: { stage: string; count: number; percentage: number }[];
  score_histogram: { range: string; count: number }[];
  age_distribution: { group: string; count: number; percentage: number }[];
  missing_data_summary: { domain: string; missing_count: number; missing_pct: number }[];
}

export interface ResourceCapacityResponse {
  config: {
    mri_daily_capacity: number;
    pet_daily_capacity: number;
    biomarker_daily_capacity: number;
  };
  mri_demand: number;
  pet_demand: number;
  biomarker_demand: number;
  mri_utilization_pct: number;
  pet_utilization_pct: number;
  biomarker_utilization_pct: number;
  mri_wait_days: number;
  pet_wait_days: number;
  disclaimer: string;
}
